src/TodoItem.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const [isEdit, setIsEdit] = useState(false);
+  const [todoValue, setTodoValue] = useState(todo.title);
+  const [shouldHandleBlur, setShouldHandleBlur] = useState(true);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleCheckbox = () => {
+    const setTodosArgs = (prevTodos: Todo[]) => {
+      return prevTodos.map((prevTodo) => {
+        if (prevTodo.id === todo.id) {
+          return { ...prevTodo, completed: !prevTodo.completed };
+        }
+
+        return prevTodo;
+      });
+    };
+
+    setTodos(setTodosArgs(todos));
+  };
+
+  const removeTodo = (id: number) => {
+    const index = todos.findIndex(elem => elem.id === id);
+
+    if (index !== -1) {
+      const newTodos = [...todos];
+
+      newTodos.splice(index, 1);
+      setTodos(newTodos);
+    }
+  };
+
+  useEffect(() => {
+    if (isEdit && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEdit]);
+
+  const handleDoubleClick = () => {
+    setShouldHandleBlur(true);
+    setIsEdit(true);
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  const handleLabelChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = event.target.value.trim();
+
+    setTodoValue(newValue === '' ? '' : event.target.value);
+  };
+
+  function updateTodo() {
+    const newTodoTitle = todoValue;
+
+    const setTodosArgs = (prevTodos: Todo[]) => {
```

Comment:
you're creating this method twice with similar logic. let's combine them into one method